src/plonk/circuit.rs

```diff
@@ -594,4 +606,70 @@ impl<F: Field> ConstraintSystem<F> {
         self.num_aux_columns += 1;
         tmp
     }
+
+    /// Hashes the `ConstraintSystem` into a `u64`.
+    pub fn hash(&self, mut hasher: &mut Blake2bState) -> [u8; 64] {
```

Comment:
Do not return a hash, just update the hasher that was provided. Also, this binding does not need to be mutable so it can just be `hasher: &mut Blake2bState`.

Reply:
i do mutate it when hashing in `Expression`s, `Column`s and `Rotation`s.